### PR TITLE
Switch HPC tests to use VIRTIO_CONSOLE

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

**Comment by pevik** [comment]
LGTM :)

File: lib/utils.pm [hunk 1]
```diff
@@ -1044,15 +1044,29 @@ sub ensure_serialdev_permissions {
...
+ 1. Execute a command that ask for a password.
+ 2. Detects password prompt
+ 3. Insert password and hits enter
+
 =cut
 sub exec_and_insert_password {
-    my $cmd = shift;
+    my ($cmd) = @_;
+    my $hashed_cmd = hashed_string("SR$cmd");
     type_string "$cmd";
```
Comment (richiejp) [inline lib/utils.pm]: You should probably wait for the shell prompt here (e.g. `wait_serial(serial_term_prompt(), undef, 0, no_regex => 1);`)

Comment (richiejp) [reply lib/utils.pm]: of course that only works on serial terminal

File: lib/utils.pm [hunk 2]
```diff
@@ -1044,15 +1044,30 @@ sub ensure_serialdev_permissions {
...
+ 1. Execute a command that ask for a password.
+ 2. Detects password prompt
+ 3. Insert password and hits enter
+
 =cut
 sub exec_and_insert_password {
-    my $cmd = shift;
+    my ($cmd) = @_;
+    my $hashed_cmd = hashed_string("SR$cmd");
+    wait_serial($self->{serial_term_prompt}, undef, 0, no_regex => 1) if is_serial_terminal();
```
Comment (asmorodskyi) [inline lib/utils.pm]: @richiejp  - something like this ?

Comment (richiejp) [reply lib/utils.pm]: you need to use the serial_terminal.pm module and import serial_term_prompt(). Similar to what you are doing for select_virtio_console().

**Comment by ldevulder** [comment]
@asmorodskyi, it's more a question than a review: but what are the benefit or virtio console? I'm wondering if this can have some benefit for HA tests too? And is it available on all arch?

**Comment by okurz** [comment]
@ldevulder it's faster, less or no problems with VNC, e.g. mistyping, but IIUC only works on qemu based backend

**Comment by asmorodskyi** [comment]
@ldevulder another good reason - developers don't like screenshots you can't select text from them :) I am getting a lot of complains every time I send openQA url to developer. VIRTIO_CONSOLE solve this problem

**Comment by okurz** [comment]
but keep in mind, as soon as someone asks you to test on bare-metal or s390x, don't cry ;)

**Comment by asmorodskyi** [comment]
I won't . This is the trick with using select_virtio_console . you just changing one variable (VIRTIO_CONSOLE=1/0) and you get back your fancy screenshots :)

**Comment by richiejp** [comment]
Technically we should be able to implement a pure serial terminal for all platforms, VIRTIO_CONSOLE is just the QEMU version. In fact it will probably be the only option for testing some embedded devices as we won't be able to get a VNC feed.

**Comment by asmorodskyi** [comment]
I was not able to finish this to make all HPC tests to pass . But current PR allows to continue work opn this in tests level. After it will be merged I can experiment directly in osd by simply changing VIRTIO_CONSOLE

**Comment by pevik** [comment]
> Technically we should be able to implement a pure serial terminal for all platforms.

@richiejp That would be great.

**Comment by asmorodskyi** [comment]
here are some proof runs which show that this PR not brake something :
http://kimball.arch.suse.de/tests/935
http://kimball.arch.suse.de/tests/932
http://kimball.arch.suse.de/tests/933
http://kimball.arch.suse.de/tests/934

**Pull request merged**